test/unit_tests/unit_test_math.cpp

```diff
@@ -114,6 +114,50 @@ namespace pvmath
         }
         return true;
     }
+
+    template <typename _Type>
+    bool pointProjection()
+    {
+        for ( uint32_t i = 0; i < Unit_Test::runCount(); ++i ) {
+            const PointBase2D<_Type> testPoint( Unit_Test::randomFloatValue<_Type>( -1000, 1000, 0.01f ), Unit_Test::randomFloatValue<_Type>( -1000, 1000, 0.01f ) );
+
+            const PointBase2D<_Type> pointBase( Unit_Test::randomFloatValue<_Type>( -1000, 1000, 0.01f ), Unit_Test::randomFloatValue<_Type>( -1000, 1000, 0.01f ) );
+            const PointBase2D<_Type> pointBaseX( Unit_Test::randomFloatValue<_Type>( -1000, 1000, 0.01f ), pointBase.y );
+            const PointBase2D<_Type> pointBaseY( pointBase.x, Unit_Test::randomFloatValue<_Type>( -1000, 1000, 0.01f ) );
+
+            const LineBase2D<_Type> lineX( pointBase, pointBaseX );
+            const LineBase2D<_Type> lineY( pointBase, pointBaseY );
+
+            const PointBase2D<_Type> resultPointX( testPoint.x, pointBase.y );
+            const PointBase2D<_Type> resultPointY( pointBase.x, testPoint.y );
+
+            if ( lineX.projection( testPoint ) != resultPointX || lineY.projection( testPoint ) != resultPointY )
```

Comment:
To fix accuracy issue we could do something like this:
`if ( pvmath::isEqual( lineX.projection( testPoint ), resultPointX, 10 ) || pvmath::isEqual( lineY.projection( testPoint ), resultPointY, 10 ) )`
by increasing accuracy threshold value by 10. Right now threshold value is 1e-5. It would be 1e-4.

Reply:
This could help. But why we could use 10 as the multiplier? It looks like a magic number.

For example, in `pointOpposition()` unit test, I get calculation error `0.0001220703125` sometimes. And to fix the accuracy issue in such way, I need to use 15 as the multiplier.
And if we had random values from -100000 to 100000, we would have to increase the threshold value even more.

Reply:
We both understand that the problem is not with the code but with IEEE754 format. One solution is to change the range of random values. Another solution is comparison with ground truth value. We could go with first solution for example.

Reply:
Hi, @ihhub.

Sorry for the late answer. 
What if the user of library will use `opposite()` method with big numbers (more than 10000)? He could find the same problem. It should be obvious for him to use `isEqual()` method instead of ==

Reply:
We cannot enhance standard float based computations so it's user's responsibility to used `float` instead of `double` if there is a need to do such.
We use epsilon value for comparison so I think end user could change this value instead.

Reply:
To have a fair results we need to use a square root of epsilon fo sigma which is `3e-5` (or 0.00003). Let me run your code on my machine and see values...

Reply:
For 90 degree angle we have `[4.37113883e-08, 1]` values of directories instead of `[0, 1]`. Value `4.37113883e-08` is standard error for float variable. The difference between possible generated points is 200. We multiply 2000 by 2 and by `4.37113883e-08` which gives us `0.0001748455532`. This explains why we cannot pass tests.

Reply:
So let's write a helper function here to compare two points with epsilon equal to 0.0002 with some explanation why we do this.

Reply:
Do you mean a helper function inside `unit_test_math.cpp`?

Reply:
Yes.

Reply:
I added :)